tsdb/compact.go

```diff
@@ -150,6 +151,9 @@ type LeveledCompactorOptions struct {
 	// PE specifies the postings encoder. It is called when compactor is writing out the postings for a label name/value pair during compaction.
 	// If it is nil then the default encoder is used. At the moment that is the "raw" encoder. See index.EncodePostingsRaw for more.
 	PE index.PostingsEncoder
+	// PD specifies the postings decoder factory to return different postings decoder based on BlockMeta. It is called when opening a block or opening the index file.
+	// If it is nil then the default decoder is used. At the moment that is the "raw" decoder. See index.DecodePostingsRaw for more.
```

Comment:
```suggestion
	// If it is nil then a default decoder is used, compatible with Prometheus v2.
```

I don't think "raw" is very meaningful here, not is it a a good idea to send the reader off to a different package.

Reply:
Updated